libchisel/src/gc.rs

```diff
@@ -0,0 +1,119 @@
+extern crate wasm_gc;
+extern crate parity_wasm;
+
+use super::ModuleTranslator;
+use parity_wasm::elements::*;
+
+pub struct WasmGC(wasm_gc::Config);
+
+impl Default for WasmGC {
+    fn default() -> WasmGC {
+        WasmGC(wasm_gc::Config::new())
+    }
+}
+
+impl WasmGC {
+    pub fn with_config(cfg : wasm_gc::Config) -> Self {
+        WasmGC(cfg)
+    }
+}
+
+macro_rules! update_section {
+        ($muts:expr, $cons:expr, $empty:expr) => {
+            match ($muts, $cons) {
+                (Some(m), Some(c)) => *m = c.clone(),
+                (Some(m), None) => *m = $empty,
+                _ => {}
+            };
+        }
+}
+
+impl ModuleTranslator for WasmGC {
+    fn translate(mut self, module: &mut Module) -> Result<bool, String> {
+        let serialized = parity_wasm::elements::serialize::<Module>(module.clone()).expect("Could not serialize module");
+        match self.0.gc(&serialized[..]) {
+            Ok(gced_bytes) => {
+                let gced = parity_wasm::elements::deserialize_buffer::<Module>(&gced_bytes[..]).expect("Could not deserialize gc'ed module");
+
+                // Presumably, the custom section will not be modified
+
+                update_section!(module.type_section_mut(), gced.type_section(), TypeSection::with_types(vec![]));
+                update_section!(module.import_section_mut(), gced.import_section(), ImportSection::with_entries(vec![]));
+                update_section!(module.function_section_mut(), gced.function_section(), FunctionSection::with_entries(vec![]));
+                update_section!(module.table_section_mut(), gced.table_section(), TableSection::with_entries(vec![]));
+                update_section!(module.memory_section_mut(), gced.memory_section(), MemorySection::with_entries(vec![]));
+                update_section!(module.global_section_mut(), gced.global_section(), GlobalSection::with_entries(vec![]));
+                update_section!(module.export_section_mut(), gced.export_section(), ExportSection::with_entries(vec![]));
+
+                // Awaiting mainlining of https://github.com/paritytech/parity-wasm/pull/250
```

Comment:
https://github.com/paritytech/parity-wasm/pull/250 was merged. Released in 0.35.3

Reply:
Please update the dependency to 0.35.3 and enable this feature

Reply:
`wasm_gc` pulls an older version and that is the one that is used, updating `Cargo.toml` has no effect.